### PR TITLE
Adding run type for role

### PR DESCRIPTION
**Pull request by mhoppal**

### CODE REVIEW

**Comment by mhoppal** [comment]
Changing the way we are setting up logrotate

**Comment by mhoppal** [comment]
Changing the way we set up logrotate do not merge yet

**Comment by tkuhlman** [comment]
This looks good to me. Craig should look at it since he wrote the run type bits for monasca-agent he may be able to spot any inconsistencies between the two.

File: tasks/configure_logrotate.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+  command: /opt/vertica/bin/vsql -U dbadmin -w {{ vertica_dbadmin_password }} -t -A -c "select NODE_NAME from NODES;"
+  sudo_user: dbadmin
+  environment: "{{vertica_env}}"
+  register: vertica_node
+
+- name: Set vertica_node name
+  set_fact: vertica_node_name={{vertica_node.stdout_lines}}
+
+- name: Place logrotate conf for vertica
+  template: src=vertica_logrotate.j2 dest=/opt/vertica/config/logrotate/mon owner=root group=root mode=644
```
Comment (craigbr) [inline tasks/configure_logrotate.yml]: Could we make the database name a parameter? Have it set to mon by default, but allow it to be overridden?

Comment (mhoppal) [inline tasks/configure_logrotate.yml]: Fixed here and in the schema, thanks for catching this and reminding me I know you mentioned this to me on Friday

File: tasks/main.yml [hunk 2]
```diff
@@ -1,19 +1,18 @@
...
-- name: Install Vertica
-  apt:  deb="{{vertica_package_location}}" state=installed
-  sudo: true
+- include: configure.yml
+  when: run_mode == 'Configure' or run_mode == 'Deploy'
 
-- include: node_users.yml
-- include: node_setup.yml
+- include: enable.yml
+  when: run_mode == 'Deploy'
```
Comment (craigbr) [inline tasks/main.yml]: Deploy should be the same as Install, then Configure, then Enable. So I believe this should be:
when: run_mode == 'Configure' or run_mode == 'Deploy'

File: tasks/enable.yml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+- name: Start vertica
```
Comment (craigbr) [inline tasks/enable.yml]: This just enables, it doesn't actually start, right? Then we should change the task name to reflect that

File: tasks/configure_logrotate.yml [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+  command: /opt/vertica/bin/vsql -U dbadmin -w {{ vertica_dbadmin_password }} -t -A -c "select NODE_NAME from NODES;"
+  sudo_user: dbadmin
+  environment: "{{vertica_env}}"
+  register: vertica_node
+
+- name: Set vertica_node name
+  set_fact: vertica_node_name={{vertica_node.stdout_lines}}
+
+- name: Place logrotate conf for vertica
+  template: src=vertica_logrotate.j2 dest=/opt/vertica/config/logrotate/{{ vertica_database_name }} owner=root group=root mode=644
```
Comment (craigbr) [inline tasks/configure_logrotate.yml]: It says this file doesn't have a newline at the end. You should add that as lack of it confuses some tools

File: defaults/main.yml [hunk 5]
```diff
@@ -3,3 +3,6 @@ vertica_catalog_dir: /var/vertica/catalog
 vertica_data_dir: /var/vertica/data
 vertica_ips: 127.0.0.1
 skip_install: false
+run_mode: Deploy
+vertica_env: {LC_ALL: en_US.UTF-8, LANG: en_US.UTF-8}
+vertica_database_name: mon
```
Comment (craigbr) [inline defaults/main.yml]: It says this file doesn't have a newline at the end. You should add that as lack of it confuses some tools

**Pull request merged**